src/rgw/rgw_op.h

```diff
@@ -1485,6 +1485,29 @@ class RGWDeleteObj : public RGWOp {
   dmc::client_id dmclock_client() override { return dmc::client_id::data; }
 };
 
+enum class TaggingPermission : short {
+NO_TAGGING_PERM,
+PUT_TAGGING_PERM,
+GET_TAGGING_PERM
+};
```

Comment:
Yes, enums aren't really for flags, at all, in spite of my suggesting it (because I hate unscoped enum).  In other places, we just use something like "constexpr static uint8_t foo1 = 0x001";  That's the nicer c++ (11?) equivalent of the preprocessor defines we would have used in the past.